src/IROperator.h

```diff
@@ -1636,6 +1636,37 @@ inline Expr count_trailing_zeros(Expr x) {
                                 {x}, Internal::Call::PureIntrinsic);
 }
 
+/** Divide two integers, rounding towards zero. This is the typical
+ * behavior of most hardware architectures, which differs from
+ * Halide's division operator, which is Euclidean (rounds towards
+ * -infinity). */
+inline Expr div_round_to_zero(Expr x, Expr y) {
```

Comment:
Yes, as-is it will infinite loop, which is why I didn't do that at first, but that's easy to fix and worth doing to get the simplifications, IMO.